azurerm/resource_arm_monitor_diagnostic_setting.go

```diff
@@ -234,6 +247,11 @@ func resourceArmMonitorDiagnosticSettingCreateUpdate(d *schema.ResourceData, met
 		valid = true
 	}
 
+	logAnalyticsDestinationType := d.Get("log_analytics_destination_type").(string)
+	if logAnalyticsDestinationType != "" {
```

Comment:
This could be simplified to
```suggestion
	if v := d.Get("log_analytics_destination_type").(string); v != "" {
		properties.DiagnosticSettings.LogAnalyticsDestinationType = &v
	}
```

Reply:
Could we also check that if a type is specified that a `log_analytics_workspace_id` is too?

Reply:
I've made the simplification and added a check that the workspaceId is set.